src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -297,12 +297,15 @@ public FormValidation doCheckRepositorySlug(@QueryParameter String value) {
         @POST
         public FormValidation doCheckServerId(@QueryParameter String serverId) {
             Jenkins.get().checkPermission(Permission.CONFIGURE);
-            List<BitbucketServerConfiguration> serverList =
-                    new BitbucketPluginConfiguration().getServerList();
+            BitbucketPluginConfiguration pluginConfiguration = new BitbucketPluginConfiguration();
+            List<BitbucketServerConfiguration> serverList = pluginConfiguration.getValidServerList();
             // Users can only demur in providing a server name if none are available to select
-            if (!serverList.isEmpty() && isBlank(serverId)) {
+            if (serverList.stream().noneMatch(server -> server.getId().equals(serverId))) {
                 return FormValidation.error("Please specify a valid Bitbucket Server.");
             }
+            if (pluginConfiguration.hasAnyInvalidConfiguration()) {
+                return FormValidation.warning("Some servers have been incorrectly configured.");
```

Comment:
I think we should also say that one or more servers have not being displayed because they are incorrectly configured